### PR TITLE
chore: [FX-3909] add Jira deployments

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-3909]

### Description

We need to connect releases (alpha, production) with Jira tasks.

### How to test

Check an extra **GH** Workflow

### Development checks

- [x] Self reviewed

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3919]: https://toptal-core.atlassian.net/browse/FX-3919?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-3909]: https://toptal-core.atlassian.net/browse/FX-3909?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

File: .github/workflows/test-jira-deployments.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Test Jira Deployments
```
Comment (dmaklygin) [inline .github/workflows/test-jira-deployments.yml:1]: remove this before merge!

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

**Review by mkrl — COMMENTED** [review]
I think it would be nice to add deployments for Storybook temploy (I think it's the PR workflow, in `deploy-docs` job). So, we will get stuff like this:

1. For alpha-package we'll track deployments to `alpha-package` environment (will be visible as "Others" in Jira)
2. For when a new Picasso version is released, we track `production` (visible as "Production" in Jira)
3. For Storybook deployments, we can track "temploy" env or something like that (will also be visible as "Others" in Jira).

Overall, this will also always include the latest links to all relevant releases at the bottom of the PR page + we can use a custom [.jira/config.yml](https://github.com/atlassian/github-for-jira/blob/main/docs/deployments.md#environment-mapping) if we want to have meaningful env names and get something like this:

- `alpha-package` => "Testing" in Jira
- `storybook-temploy` => "Testing" in Jira
- `production-package` => "Production" in Jira
- `production-storybook` => "Production" in Jira

This way, we'll have maximum transparency and access to all relevant deployments in every ticket and at the bottom of the PR.

**Review by TomasSlama — APPROVED** [review]

**Comment by ic-latpot** [comment]
!MTGL

**Comment by dmaklygin** [comment]
> @dmaklygin with this image We can probably even remove the old message image

Good point. Let's remove it

**Review by augustobmoura — APPROVED** [review]

**Pull request merged**